app/src/main/java/com/github/sdpsharelook/section/CardAdapter.kt

```diff
@@ -6,21 +6,25 @@ import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import com.github.sdpsharelook.databinding.CardSectionBinding
+import com.github.sdpsharelook.storage.RTDBWordListRepository
+import dagger.hilt.android.AndroidEntryPoint
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class CardAdapter(
-    private val sections: List<Section>,
+class CardAdapter constructor(
     private val clickListener: SectionClickListener,
-    private val dialog: Dialog
+    private val dialog: Dialog,
+    private val wordRTDB : RTDBWordListRepository
```

Comment:
Should still be an interface typed variable
```suggestion
    private val wordRepo: IRepository<List<Word>>
```